components/resource-creation/ResourceDisplay.tsx

```diff
@@ -103,10 +130,33 @@ function ResourceDisplay() {
     if (id && selectedPatient) {
       const resourceIndexToDelete = currentTestCases[selectedPatient].resources.findIndex(r => r.id === id);
       if (resourceIndexToDelete >= 0) {
-        const nextResourceState = produce(currentTestCases, draftState => {
+        setIsCalculationLoading(true);
+        produce(currentTestCases, async draftState => {
           draftState[selectedPatient].resources.splice(resourceIndexToDelete, 1);
+          // re-run measure report calculations for updated state
+          if (measureBundle.content) {
```

Comment:
it seems like this conditional w/ try/catch is used multiple times. Can we break that out into a function?